homeassistant/components/knx/light.py

```diff
@@ -38,13 +40,89 @@ async def async_setup_platform(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up lights for KNX platform."""
+    _async_migrate_unique_id(hass, discovery_info)
     entities = []
     for device in hass.data[DOMAIN].xknx.devices:
         if isinstance(device, XknxLight):
             entities.append(KNXLight(device))
     async_add_entities(entities)
 
 
+@callback
+def _async_migrate_unique_id(
+    hass: HomeAssistant, discovery_info: DiscoveryInfoType | None
+) -> None:
+    """Change unique_ids used in 2021.4 to exchange individual color switch address for brightness address."""
+    entity_registry = er.async_get(hass)
+    if not discovery_info or not discovery_info["platform_config"]:
```

Comment:
A future improvement would be to not load the platform at all unless it has `platform_config` and move the guard clause for `discovery_info` being `None` to the beginning of `async_setup_platform`. We can then tighten down the parameter typing in `_async_migrate_unique_id` and not load platforms if not needed.

Reply:
this is done in #49275
just like for climate and cover I could then just use 
```python
@callback
 def _async_migrate_unique_id(
     hass: HomeAssistant, platform_config: list[ConfigType]
 ) -> None:
```